### PR TITLE
fix(manager): resume block production right when skew < max skew

### PR DESCRIPTION
**Pull request by srene**
# PR Standards

## Opening a pull request should be able to meet the following requirements

--

PR naming convention: https://hackmd.io/@nZpxHZ0CT7O5ngTp0TP9mg/HJP_jrm7A

---

Close #XXX

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/produce.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context, bytesProducedC chan int)
 				return nil
 			case bytesProducedC <- bytesProducedN:
 			default:
-				evt := &events.DataHealthStatus{Error: fmt.Errorf("bytes produced channel is full: %w", gerrc.ErrResourceExhausted)}
+				evt := &events.DataHealthStatus{Error: fmt.Errorf("Block production paused. Time between last block produced and last block submitted higher than max skew time %s: %w", m.Conf.MaxSkewTime, gerrc.ErrResourceExhausted)}
```
Comment (omritoptix) [inline block/produce.go:85]: probably worth also putting here the last block subimtted time

Comment (srene) [reply block/produce.go]: done

File: block/submit.go [hunk 2]
```diff
@@ -110,6 +110,8 @@ func SubmitLoopInner(
 				UpdateBatchSubmissionGauges(pending, unsubmittedBlocksNum(), batchSkewTime())
 
 				if done || nothingToSubmit || (lastSubmissionIsRecent && maxDataNotExceeded) {
+					pendingBytes.Store(pending)
+					ticker.Reset(maxBatchSubmitTime)
```
Comment (omritoptix) [inline block/submit.go:114]: I don't understand this. can you please explain? thanks

Comment (srene) [reply block/submit.go]: moved after loop and documented.

File: block/submit.go [hunk 3]
```diff
@@ -128,11 +127,14 @@ func SubmitLoopInner(
 					}
 					return err
 				}
-				ticker.Reset(maxBatchSubmitTime)
-				pending = uint64(unsubmittedBlocksBytes())                                                                                 //nolint:gosec // bytes size is always positive
-				logger.Info("Submitted a batch to both sub-layers.", "n bytes consumed from pending", nConsumed, "pending after", pending) // TODO: debug level
+				pending = uint64(unsubmittedBlocksBytes()) //nolint:gosec // bytes size is always positive
+				if batchSkewTime() < maxSkewTime {
```
Comment (mtsitrin) [inline block/submit.go:131]: can u elaborate on the use of this `if`?
why not call `trigger.Nudge()` after submission loop finished as before?

**Pull request merged**